Update ddclient.in to re-support Sitelutions.com

scoutdriver73 opened the pull request:
Update sitelutions protocol to support their APIv2. Other than cosmetic notes/changes, the only real code change is the server changes from www.sitelutions.com to api2.sitelutions.com

It would be trivial to URL encode the API key for the user, but unsure what the end-user would be expecting (assuming they read the documentation). 

https://sitelutions.com/index.php?rp=/knowledgebase/4/Setting-up-Dynamic-DNS.html

scoutdriver73 committed:
Update ddclient.in

Update sitelutions protocol to support their APIv2.
Other than cosmetic notes/changes, the only real is the server changes from www.sitelutions.com to api2.sitelutions.com

indrajitr commented on ddclient.in:
```diff
@@ -5118,10 +5118,10 @@ EoEXAMPLE
-## https://www.sitelutions.com/dnsup?id=990331&user=myemail@mydomain.com&pass=SecretPass&ip=192.168.10.4
-## domain=domain.com&password=domain_password&ip=your_ip
+## https://api2.sitelutions.com/dnsup?id=990331&user=myemail@mydomain.com&pass=urlencodedAPIkey&ip=192.168.10.4
+## domain=domain.com&password=URL_encoded_API_key&ip=your_ip
```
Do we need both `pass` and `password`? Second `password` is probably superfluous.

indrajitr commented on ddclient.in:
```diff
@@ -1229,7 +1229,7 @@ our %protocols = (
             %{$cfgvars{'protocol-common-defaults'}},
-            'server'       => setv(T_FQDNP, 0, 'www.sitelutions.com', undef),
+            'server'       => setv(T_FQDNP, 0, 'api2.sitelutions.com', undef),
             'min-interval' => setv(T_DELAY, 0, interval('5m'), interval('5m')),
```
Can you please add `ttl` as option here? api2.sitelutions.com supports this param.

indrajitr reviewed (changes requested):
We can proceed with this with some small changes. Assuming you have tested this yourself.

Also, can you please rebase with the current `main` branch?

beadon committed:
sitelutions: update to APIv2 and add ttl option

Change the default server from www.sitelutions.com to api2.sitelutions.com
to use the current Sitelutions APIv2 endpoint. Add an optional ttl
configuration variable supported by the new API.

Also clean up the examples and update comments to match the APIv2 URL format.
